implementations/code_df/code_changes_git.py

```diff
@@ -1,152 +1,52 @@
-        from the "since" date to the "until" date arguments, specified
-        during object initiation.
-
-        The fixed time interval can be either a month or a week.
-
-        :param period: A string which can be either "month" or "week"
+        Count number of commits of different types, like including
+        empty commits or counting only those commits made on
+        the master branch.
 
-        :returns dataframe: A DataFrame whose rows each represent an interval
-            of "period" and the count for that interval
+        :returns count: the number of commits satisfying the conditions passed
+            while instantiating CodeChangesGit.
         """
 
-        df = self.df
-        if period == 'month':
-            timeseries_series = df['created_date'] \
-                .groupby([df['created_date'].dt.year.rename('year'),
-                          df['created_date'].dt.month.rename('month')]) \
-                .agg('count')
+        return len(self.df['hash'].unique())
 
-            all_periods = pd.DataFrame(
-                            pd.date_range(self.since, self.until, freq='M'),
-                            columns=['Dates'])
-            all_periods = pd.DataFrame(
-                [all_periods['Dates'].dt.year.rename('year'),
-                 all_periods['Dates'].dt.month.rename("month")]).T
-
-        elif period == 'week':
-            timeseries_series = df['created_date'] \
-                .groupby([df['created_date'].dt.year.rename('year'),
-                          df['created_date'].dt.week.rename('week')])   \
-                .agg('count')
-
-            all_periods = pd.DataFrame(
-                            pd.date_range(self.since, self.until, freq='W'),
-                            columns=['Dates'])
-            all_periods = pd.DataFrame(
-                [all_periods['Dates'].dt.year.rename('year'),
-                 all_periods['Dates'].dt.week.rename('week')]).T
-
-        else:
-            raise ValueError("period parameter can take 'month' or 'week'")
-
-        timeseries_df = pd.DataFrame(timeseries_series)
-        timeseries_df.reset_index(inplace=True)
-        timeseries_df.columns = ['year', period, 'count']
-        merged_df = all_periods.merge(timeseries_df, how='outer').fillna(0)
-
-        dataframe = merged_df
-        return dataframe
-
-    def _count_master_only(self, incl_empty=True):
+    @utils.rename_function('count')
+    def _aggregate(self, series):
         """
-        Counts commits present only on the master branch.
+        Perform an aggregation operation on a DataFrame or Series.
 
-        :param incl_empty: exclude empty commits on the master branch
+        :param series: A pandas.Series object passed implicitly during
+        aggregation operations.
 
-        :returns commits_count: the number of commits created on the
-            master branch
+        :returns: aggregation value, like mean, sum or count.
```

Comment:
But it seems this always counts... When is it computing other stuff, like mean or sum?

Reply:
Will make that more specific.